x-pack/plugins/security_solution/public/management/cypress/e2e/endpoint/endpoints.cy.ts

```diff
@@ -5,15 +5,90 @@
  * 2.0.
  */
 
+import type { Agent } from '@kbn/fleet-plugin/common';
+import { ENDPOINT_VM_NAME } from '../../tasks/common';
+import {
+  getAgentByHostName,
+  getEndpointIntegrationVersion,
+  reassignAgentPolicy,
+} from '../../tasks/fleet';
+import type { IndexedFleetEndpointPolicyResponse } from '../../../../../common/endpoint/data_loaders/index_fleet_endpoint_policy';
 import { login } from '../../tasks/login';
+import {
+  AGENT_HOSTNAME_CELL,
+  TABLE_ROW_ACTIONS,
+  TABLE_ROW_ACTIONS_MENU,
+  AGENT_POLICY_CELL,
+} from '../../screens/endpoints';
+import {
+  FLEET_REASSIGN_POLICY_MODAL,
+  FLEET_REASSIGN_POLICY_MODAL_CONFIRM_BUTTON,
+} from '../../screens/fleet';
 
 describe('Endpoints page', () => {
+  const endpointHostname = Cypress.env(ENDPOINT_VM_NAME);
+
   beforeEach(() => {
     login();
   });
 
   it('Loads the endpoints page', () => {
     cy.visit('/app/security/administration/endpoints');
     cy.contains('Hosts running Elastic Defend').should('exist');
+    cy.getByTestId(AGENT_HOSTNAME_CELL).should('have.text', endpointHostname);
+  });
+
+  describe('Endpoint reassignment', () => {
+    let response: IndexedFleetEndpointPolicyResponse;
+    let initialAgentData: Agent;
+
+    before(() => {
+      getAgentByHostName(endpointHostname).then((agentData) => {
+        initialAgentData = agentData;
+      });
+      getEndpointIntegrationVersion().then((version) => {
+        cy.task<IndexedFleetEndpointPolicyResponse>('indexFleetEndpointPolicy', {
+          policyName: `Reassign ${Math.random().toString(36).substr(2, 5)}`,
+          endpointPackageVersion: version,
+        }).then((data) => {
+          response = data;
+        });
+      });
+    });
+
+    beforeEach(() => {
+      login();
+    });
+
+    after(() => {
+      if (initialAgentData?.policy_id) {
+        reassignAgentPolicy(initialAgentData.id, initialAgentData.policy_id);
+      }
+      if (response) {
+        cy.task('deleteIndexedFleetEndpointPolicies', response);
+      }
+    });
+
+    it('User can reassign a single endpoint to a different Agent Configuration', () => {
+      cy.visit('/app/security/administration/endpoints');
```

Comment:
instead of hardcoding this, can you use the routing utilities that we have here to get the path:

https://github.com/elastic/kibana/blob/14cd9ce0baf7d9e6c1c088abac059086e5a3f7fd/x-pack/plugins/security_solution/public/management/common/routing.ts